recipes/thrift/all/conanfile.py

```diff
@@ -0,0 +1,146 @@
+import os
+from conans import tools, CMake, ConanFile
+from conans.errors import ConanInvalidConfiguration
+
+
+class ConanFileDefault(ConanFile):
+    name = "thrift"
+
+    description = "Thrift is an associated code generation mechanism for RPC"
+    topics = ("conan", "thrift", "serialization", "rpc")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apache/thrift"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_libevent": [True, False],
+        "with_openssl": [True, False],
+        "with_boost_functional": [True, False],
+        "with_boost_smart_ptr": [True, False],
+        "with_boost_static": [True, False],
+        "with_boostthreads": [True, False],
+        "with_stdthreads": [True, False],
+        "with_c_glib": [True, False],
+        "with_cpp": [True, False],
+        "with_java": [True, False],
+        "with_python": [True, False],
+        "with_haskell": [True, False],
+        "with_plugin": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "with_libevent": True,
+        "with_openssl": True,
+        "with_boost_functional": False,
+        "with_boost_smart_ptr": False,
+        "with_boost_static": False,
+        "with_boostthreads": False,
+        "with_stdthreads": True,
+        "with_c_glib": False,
+        "with_cpp": True,
+        "with_java": False,
+        "with_python": False,
+        "with_haskell": False,
+        "with_plugin": False
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        # See: https://github.com/apache/thrift/blob/v0.12.0/build/cmake/DefinePlatformSpecifc.cmake
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Thrift does not currently support shared libs on Windows. Use -o thrift:shared=False instead")
```

Comment:
Windows shared is now supported by upstream

https://github.com/apache/thrift/blame/v0.13.0/build/cmake/DefinePlatformSpecifc.cmake#L51

Reply:
Or at least it seems so. I can't get it to work locally

Reply:
it works with `-o openssl:shared=True`, I'm exploring patching cmake files